tests/pass/libc.rs

```diff
@@ -290,10 +290,28 @@ fn test_clocks() {
     assert_eq!(is_error, 0);
 }
 
+fn test_posix_gettimeofday() {
+    let mut tp = std::mem::MaybeUninit::<libc::timeval>::uninit();
+    let tz = std::ptr::null::<libc::timezone>();
+    #[cfg(target_os = "macos")]
+    let tz = tz as *mut libc::c_void;
+    let is_error = unsafe { libc::gettimeofday(tp.as_mut_ptr(), tz as *mut _) };
+    assert_eq!(is_error, 0);
```

Comment:
Can you read the field of `tp` for some basic sanity checks? Like, at least reading them into local variables should be allowed (i.e., they have actually been initialized).